src/Model/Operation/ProductSyncHandler.php

```diff
@@ -274,6 +286,32 @@ private function handleConfiguratorGroups(SalesChannelProductEntity $product): a
         return $mainProducts;
     }
 
+    private function handleFirstActiveVariant(ProductEntity $product): array
+    {
+        $variants = [];
+        foreach ($product->getChildren() as $child) {
+            $variants[$child->getId()] = $child;
+        }
+
+        $mainProducts = [];
+        foreach ($variants as $key => $variant) {
+            if ($variant->active) {
+                $mainProduct = $variant;
+                unset($variants[$key]);
+                break;
+            }
+        }
```

Comment:
`$product->getChildren()` has multiple function this can be replaced with. Best would probably be to use `filterByProperty` and `first()` combined.